libcomps/src/comps_parse.c

```diff
@@ -218,6 +219,7 @@ signed char comps_parse_file(COMPS_Parsed *parsed, FILE *f,
         buff = XML_GetBuffer(parsed->parser, BUFF_SIZE);
         if (buff == NULL) {
             comps_log_error(parsed->log, COMPS_ERR_MALLOC, 0);
+            fclose(f);
```

Comment:
It is strange to me that the function that did not call `fopen` calls `fclose`. But the patch is correct.

Reply:
Yes, that's not nice. But the comps_parse_file () function is in the header file. And I only fix resource leaks. I don't want to change the behavior of the function (API).